safekeeper/src/safekeeper.rs

```diff
@@ -183,12 +183,17 @@ pub struct SafeKeeperState {
     /// for correctness, exists for monitoring purposes.
     #[serde(with = "hex")]
     pub proposer_uuid: PgUuid,
+    /// Since which LSN this timeline generally starts. Safekeeper might have
+    /// joined later.
+    pub timeline_start_lsn: Lsn,
+    /// Since which LSN safekeeper has (had) WAL for this timeline.
+    pub local_start_lsn: Lsn,
```

Comment:
I wonder should we update this LSN once we trim WAL locally? If not, how we should get the lowest LSN that is truly available on disk? Is it `max(local_start_lsn, last_removed_segno * wal_seg_size)`?

Reply:
Do we really need that? We can
1) check `get_horizon_segno` to know LSN which is reliably available;
2) everything below can be missing. Obviously we can check file existence on FS, but not sure why would we want that.

Reply:
My worries were mostly about trying to read non-existent WAL somehow. And I thought to check against some LSN before every read. But now I understood that we would get an error when reading already deleted file, so the only bad case is reading zeroes right before `local_start_lsn`, so now I think we don't really need anything additional here.

P.S. We probably should write somewhere that the only valid case for incomplete segment on disk (with zeroes filled prefix) is one that contains `local_start_lsn`. Every other segment without ".partial" suffix should be considered as having valid WAL from start to finish.